src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelGene.tsx

```diff
@@ -97,10 +103,8 @@ const TrackPanelGene = (props: TrackPanelGeneProps) => {
 
   // set status of all transcripts based on the saved redux state after loading component
   useEffect(() => {
-    if (allTranscriptsInGene?.length) {
-      updateFocusGeneTranscripts(visibleTranscriptIds);
-    }
-  }, [allTranscriptsInGene]);
+    updateFocusGeneTranscripts(visibleTranscriptIds);
+  }, [geneId, visibleTranscriptIds?.length]);
```

Comment:
@azangru `visibleTranscriptIds` can change despite being the same length, so your fix with the stringified array would be more suitable here.

Reply:
I was thinking that there's no way on the React side to change visibleTranscriptIds without also changing the visibleTranscriptIds.length; because you have to toggle the transcripts one at a time, which will change the length of the array. But comparing stringified sorted arrays does feel more robust, so I made the change and pushed it.